### PR TITLE
adding a reverse_action function

### PR DESCRIPTION
**Pull request by CamDavidsonPilon**
I saw a few instances where this function could be useful (I found it useful implementing my v1 of a person IPD project.). There are probably some more instances in other strategies. 

Couple things: 1) I'm so-so on the name `reverse_action` - I'm happy to hear other suggestions. 2) This changes the implementation of TitFotTat to something I think it more readable. 

Thoughts?

### CODE REVIEW

File: axelrod/strategies/titfortat.py [hunk 1]
```diff
@@ -18,7 +17,9 @@ class TitForTat(Player):
 
     @staticmethod
     def strategy(opponent):
-        return D if opponent.history[-1:] == [D] else C
+        if len(opponent.history) == 0:
+            return C
+        return reverse_action(opponent.history[-1])
```
Comment (CamDavidsonPilon) [inline axelrod/strategies/titfortat.py]: hmm, something about this is causing lots of tests to fail - the logic looks identical to me however.

**Comment by drvinceknight** [comment]
I haven't looked too closely. 

I'm hesitant (not against, just hesitant) about the readability claim of Tit for Tat. I think having it written in base python makes it an accessible example when describing the library to non python/coders.

**Comment by drvinceknight** [comment]
Weird about the failing tests. On my phone, will take a closer look when in front of a machine.

**Comment by meatballs** [comment]
> I think having it written in base python makes it an accessible example when describing the library to non python/coders.

I agree. We know there are some very inexperienced coders looking at this library and we need to ensure that we have some easy routes in - even if that sometimes leads to code that's not quite as DRY as it could be.

**Comment by drvinceknight** [comment]
> I agree. We know there are some very inexperienced coders looking at this library and we need to ensure that we have some easy routes in - even if that sometimes leads to code that's not quite as DRY as it could be.

I would suggest (forgive me if I'm speaking incorrectly on your behalf @meatballs) that we both think this is a nice contribution and good to have but perhaps `TitForTat` should be left alone. @marcharper has suggested that we write `TFT` in an even more verbose basic python way than it is now which I don't think is a bad idea...

**Comment by drvinceknight** [comment]
Not too sure what's happening with the tests, travis is saying something about numpy but running them locally I got other errors. For info this is how to run the unit tests @CamDavidsonPilon:

```
cd Axelrod
python -m unittest discover axelrod.tests.unit
```

**Comment by meatballs** [comment]
> I would suggest (forgive me if I'm speaking incorrectly on your behalf @meatballs) that we both think this is a nice contribution and good to have but perhaps TitForTat should be left alone.

Exactly my thoughts.

**Comment by marcharper** [comment]
While I agree re: TFT, it is somewhat silly to have `flip_dict` defined in several places, so I'm for the idea behind this PR. Perhaps `toggle_action` is a better name.

My suggestion for TFT rather than:

```
def strategy(opponent):
    return 'D' if opponent.history[-1:] == ['D'] else 'C'
```
We use something (rather verbose) like:
```
def strategy(opponent):
    # First move
    if len(self.history) == 0:
        return C
    # React to the opponent's last move
    if opponent.history[-1] == D:
        return D
    return C
```

The reason is that slicing is not something I would expect a python beginner to understand. In any case, perhaps we should put in a comment "Don't mess with TFT, thanks!" in the code.

**Comment by drvinceknight** [comment]
> In any case, perhaps we should put in a comment "Don't mess with TFT, thanks!" in the code.

I like both of the above suggestion (the code for TitForTat) and the comment in TitForTat: "This strategy is written in a more verbose way than necessary however it is left that way as an example for contributors who are perhaps less familiar with Python" or something like that...

**Comment by CamDavidsonPilon** [comment]
:ok_hand: I've removed changes to TFT, and tests pass locally. Waiting for :green_apple:

**Comment by marcharper** [comment]
Any votes for `toggle_action`? `reverse_action` sounds more like rolling back than swapping. More suggestions: `swap_action`, `flip_action`, `transpose_actions`...

**Comment by drvinceknight** [comment]
I think I prefer `flip_action`, but `toggle_action` would be my second favourite. I agree that `reverse_action` is not ideal, sounds like some sort of manipulation.

**Comment by meatballs** [comment]
```flip_action``` for me

**Comment by marcharper** [comment]
Looks like numpy is failing to install on Travis

**Comment by drvinceknight** [comment]
This happened the other day. I'll restart it. Am I the only person who can
restart a travis run? We should fix that if so.

**Comment by marcharper** [comment]
I thought so but it looks like I can do it (had to sign in through github).

**Comment by drvinceknight** [comment]
Cool :)

**Comment by drvinceknight** [comment]
Is this good to go now?

**Comment by meatballs** [comment]
:+1:

**Comment by drvinceknight** [comment]
Let's wait to make sure @marcharper is ok with `flip_action` as he's been involved with the review :tada: :shipit:

**Pull request merged**